### DISCUSSION TITLE
Datadog deployment sync

### DISCUSSION THREAD

**Discussion by nightmareartist** [argoproj/argo-cd#6364] [Q&A] [answered]
Hi,

I have setup Datadog using ArgoCD and things seem to work just fine. I have an app pointing to my GitHub repo that contains `values.yaml` and Datadog charts. It is not something I change very often. However, whenever anything else is deployed in the cluster through ArgoCD (other services) Datadog is being re-deployed too. Every single time. When I go into Datadog app in ArgoCD I see that the sync is actually tied to changes made in other apps, "To" link in ArgoCD points to a repo of another app. However, nothing changed for Datadog. I am trying to understand why ArgoCD thinks that Datadog is out of sync and it needs to be re-deployed even though no changes were made on Datadog config files.

Thank you in advance.

**Comment by alexmt** [comment, marked as answer]
This might be happening because some manifest in Datadog chart uses functions like `now` and so every manifest generation produces slightly different results.

Argo CD assumes that the chart stored in Git generates the same manifests and regenerate manifests only when a new commit is detected. You can try to click "Hard refresh" button on Datadog app and see which part of the manifest changes.

**Reply by argychatzi** [reply]
Thanks for your response @alexmt. This is correct, we verified that they use helm function `randAlphaNum` to generate a token and store it in a secret, which means every time `helm template` is invoked, argo automatically marks the app as `OutOfSync`. Looking into their documentation, they also provide the possibility to specify that particular token either directly as clear text in the values file or via referencing a secret created outside their helm chart.

**Reply by vl-kp** [reply]
@argychatzi any temp solution for it?

**Reply by nightmareartist** [reply]
@vl-kp You can fix this permanently by using an existing token in an external secret. The way I've fixed it in GCP is along these lines:

1. Generate a random 32 char token i.e. `openssl rand -hex 32`
2. Put that token as a secret in the Secret manager i.e. `{ "token" : "the-random-32-token" }`
3. Use External Secret Operator (https://external-secrets.io/) to pull that token into a K8s secret that we can call `datadog-secret`
4. Edit `values.yaml` and add into it the following:
```
    clusterAgent:
        tokenExistingSecret: "datadog-secret"
```
This is commented [here](https://github.com/DataDog/helm-charts/blob/baed3e51ed3d90b13b0d26ff3b97fbb17e6e0988/charts/datadog/values.yaml#L837).

The issue is now gone. You can do something similar in AWS or any other cloud service.

**Reply by vl-kp** [reply]
Thanks for the prompt reply and informative answer
